packages/ocp-build/ocp-build.1.99.10-beta/opam

```diff
@@ -23,6 +23,8 @@ remove: [
   [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-subcmd" ]
   [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-system" ]
   [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-unix" ]
+  [ "rm" "-f" "%{prefix}%/lib/ocaml/typerex/installed.ocp" ]
+  [ "rmdir" "--ignore-fail-on-non-empty" "%{prefix}%/lib/ocaml/typerex" ]
```

Comment:
Is `ocp-build` the owner of this directory? If so, `rmdir` alone might be fine or `rm -r` if you'd like to clean up for other delinquent users. Otherwise, something like

```
[ "sh" "-exc" "rmdir %{prefix}%/lib/ocaml/typerex || true" ]
```

would be more portable but could "succeed" even when the errno is not `ENOTEMPTY` but that would be very similar observationally to what you have now on GNU/Linux.

I recommend having a straightforward path ownership discipline (if you don't already) for your packages and then shared resource removal becomes simple.